iothub/device/src/Common/Exceptions/DeviceNotFoundException.cs

```diff
@@ -8,7 +8,19 @@
 namespace Microsoft.Azure.Devices.Client.Exceptions
 {
     /// <summary>
-    /// The exception that is thrown when an attempt to communicate with a device fails because the given device identifier cannot be found.
+    /// The exception is thrown when the device is disabled and will be used to set the status to device disabled in the
+    /// connection status handler. This exception also corresponds to the following error codes on operation responses:
+    /// <list>
+    /// <item>
+    /// AmqpErrorCode.NotFound
+    /// </item>
+    /// <item>
+    /// HttpStatusCode.NotFound
+    /// </item>
+    /// <item>
+    /// HttpStatusCode.NoContent
```

Comment:
In case you raise an eyebrow at this, @vinagesh is just documenting what we currently do. I haven't the foggiest idea why we'd treat a 204 (NoContent) as a DeviceNotFoundException, though

Reply:
yes unfortunately I don't know why we do this but too late to change anyway.